containers/src/Data/IntMap/Internal.hs

```diff
@@ -324,11 +324,11 @@ import Data.Data (Data(..), Constr, mkConstr, constrIndex,
 import qualified Data.Data as Data
 import GHC.Exts (build)
 import qualified GHC.Exts as GHCExts
-import Text.Read
 import Language.Haskell.TH.Syntax (Lift)
 -- See Note [ Template Haskell Dependencies ]
 import Language.Haskell.TH ()
 #endif
+import Text.Read
```

Comment:
We should use `#if defined(__GLASGOW_HASKELL__) || defined(__MHS__)` for this, to reflect the Read instance.